libraries/oneview_helper.rb

```diff
@@ -28,21 +28,26 @@ def self.do_resource_action(context, type, action)
     # @param context Context from the resource action block (self)
     # @param [String, Symbol] type Name of the resource type. e.g., :EthernetNetwork
     # @return [Class] Resource class
-    def self.get_resource_class(context, resource_type)
+    def self.get_resource_class(context, resource_type, secondary_api = nil)
       load_sdk(context)
       api_version = context.property_is_set?(:api_version) ? context.api_version : context.node['oneview']['api_version']
-      api_module = get_api_module(api_version)
+      api_module = get_api_module(api_version, secondary_api)
       api_variant = context.property_is_set?(:api_variant) ? context.api_variant : context.node['oneview']['api_variant']
       api_module.provider_named(resource_type, api_variant)
     end
 
     # Get the API module given an api_version
     # @param [Fixnum, String] api_version
+    # @param [Symbol, String] secondary_api The additional API supported inside OneviewCookbook. e.g. ImageStreamer
     # @return [Module] Resource module
-    def self.get_api_module(api_version)
-      OneviewCookbook.const_get("API#{api_version}")
+    def self.get_api_module(api_version, secondary_api = nil)
+      base_sdk = secondary_api ? OneviewCookbook.const_get(secondary_api) : OneviewCookbook
+      base_sdk.const_get("API#{api_version}")
     rescue NameError
-      raise NameError, "The api_version #{api_version} is not supported. Please use a supported version."
+      msg = "The api_version #{api_version} is not supported"
+      msg += secondary_api ? " in #{secondary_api}" : '.'
```

Comment:
Same here with the param name, docs, and default. This will simplify the method too.